README.md

```diff
@@ -110,6 +109,31 @@ end
 T::Coerce[Param].new.from({id: '1'})
 # => <Param id=1, role="wizard">
 ```
+More examples: [nested params](https://github.com/chanzuckerberg/sorbet-coerce/blob/a56c0c6a363bb49b11e77ac57893afc3d54c6b8c/spec/nested_spec.rb#L18-L26)
+
+## Coercion Error
+
+Sorbet-coerce throws a coercion error when it fails to convert a value into the specified type. The error is [configurable](https://sorbet.org/docs/runtime#changing-the-runtime-behavior) through `T::Configuration`. In an environment where type errors are configured to be silent (referred to soft errors), when the coercion fails (or constructing `T::Struct` fails), `T::Coerce` will return the original value instead of actually raising the errors (referred to hard errors).
+
+## `null`, `''`, and `undefined`
+
+Sorbet-coerce is designed in the context of web development. When coercing into a `T::Struct`, the values that need to be coerced are often JSON-like. Suppose we're coercing object `json` into a `Params` instance
+```ruby
+json = {"a": "1", "null_filed": null, "blank_filed": ""}
+
+class Params < T::Struct
+  const :a, Integer
+  const :null_filed, T.nilable(Integer)
+  const :blank_filed, T.nilable(Integer)
+  const :missing_key, T::Array[Integer], default: []
+end
+
+params = T::Coerce[Params].new.from(json)
```

Comment:
add the value of params here so that people can see what it'll coerce into

Reply:
Good point!